common/src/lib.rs

```diff
@@ -326,6 +356,44 @@ pub fn remove_waiting_message(prog_id: H256, msg_id: H256) -> Option<(Message, u
     msg
 }
 
+pub struct WaitingMessageIterator {
```

Comment:
Add an issue to optimise it without iterator please

Reply:
done = #525

Reply:
There is also this "built-in" `frame_support::storage::PrefixIterator`, that can be easily customized for any prefix (in case you find it more convenient). We have examples in our code.